topic_manager.go

```diff
@@ -229,6 +229,38 @@ func (m *topicManager) ensureExists(topic string, npar, rfactor int, config map[
 	return nil
 }
 
+func (m *topicManager) waitForCreated(topic string) error {
+
+	// no timeout defined -> no check
+	if m.topicManagerConfig.CreateTopicTimeout == 0 {
+		return nil
+	}
+
+	start := time.Now()
```

Comment:
Another approach
```
// or better get ctx passed as param in waitForCreated instead of using .Background()
ctx, cancel := context.WithTimeout(context.Background, m.topicManagerConfig.CreateTopicTimeout)
defer cancel()

for ctx.Err() == nil {
	_, err := m.Partitions(topic)
	switch err {
	case nil: 
		return nil
	case errTopicNotFound:
		time.Sleep(time.Second)
	default: 
		return fmt.Errorf("error checking topic: %w", err)
	}
}
```

Reply:
I was thinking of adding the context as a param, but that would mean changing the whole api.
But the timeout context is definitely an improvement, didn't think of this. so Thanks :)

Reply:
@paulhenke fixed.